### PR TITLE
Increase Testing

### PR DESCRIPTION
**Pull request by pixelcircuits**
note: also includes the compression refactor

### CODE REVIEW

File: test/foundry/TransferEth.t.sol [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        //verify end state
+        uint256 solverTokenBalance = _testERC20.balanceOf(_publicAddressSolver);
+        assertEq(solverTokenBalance, erc20ReleaseAmount, "The solver ended up with incorrect token balance");
+
+        uint256 userTokenBalance = _testERC20.balanceOf(address(_account));
+        uint256 expectedUserTokenBalance = accountInitialERC20Balance - erc20ReleaseAmount;
+        assertEq(userTokenBalance, expectedUserTokenBalance, "The user ended up with incorrect token balance");
+
+        uint256 recipientBalance = address(_recipientAddress).balance;
+        assertEq(recipientBalance, transferAmount, "The recipient didn't get the expected ETH");
```
Comment (supiket) [inline test/foundry/TransferEth.t.sol:33]: GitHub thinks the apostrophe here finishes the string that starts with the quotation mark. In the PR as well as the regular code viewer, it looks like everything else that is following the apostrophe is a string. This is not actually the case, so we can ignore this, add an escape for the apostrophe, or get rid of the contraction.

File: src/samples/compression/EntryPointCompression.sol [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            mstore8(add(len, 27), 32)
+            mstore(add(ptr, 0x0380), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore8(add(len, 28), 32)
+            mstore(add(ptr, 0x03a0), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore8(add(len, 29), 32)
+            mstore(add(ptr, 0x03c0), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore8(add(len, 30), 32)
+            mstore(add(ptr, 0x03e0), 0x0000000000000000000000000000000000000000000000000000000000000000)
+            mstore8(add(len, 31), 32)
+            */
```
Comment (supiket) [inline src/samples/compression/EntryPointCompression.sol:100]:
```suggestion
```

Comment (pixelcircuits) [reply src/samples/compression/EntryPointCompression.sol:100]: I've kept this commented to show that other implementations can take advantage of more entries

Comment (supiket) [reply src/samples/compression/EntryPointCompression.sol:100]: Ohhh got it

**Review by supiket — APPROVED** [review]
Did a second pass of the compression, looks great

**Pull request merged**